File: src/main/java/io/mseemann/medium/jmxpg/AppStateSimulation.java
package io.mseemann.medium.jmxpg;

import io.mseemann.medium.jmxpg.beans.AppInfo;
import lombok.SneakyThrows;

import java.time.Duration;

public class AppStateSimulation implements Runnable {

    private final AppInfo appInfo;

    public AppStateSimulation(AppInfo appInfo) {
        this.appInfo = appInfo;
    }

    @SneakyThrows
    @Override
    public void run() {
        while (true) {
            Thread.sleep(Duration.ofSeconds(10).toMillis());
            String newState = appInfo.getStatus().equals("running") ? "failed" : "running";
            appInfo.setStatus(newState);
        }
    }
}

File: src/main/kotlin/io/mseemann/medium/jmxpg/AppStateSimulation.kt
package io.mseemann.medium.jmxpg

import io.mseemann.medium.jmxpg.beans.AppInfo
import java.time.Duration

class AppStateSimulation(private val appInfo: AppInfo) : Runnable {

    override fun run() {
        while (true) {
            Thread.sleep(Duration.ofSeconds(10).toMillis())
            appInfo.status = if (appInfo.status == "running") "failed" else "running"
        }
    }

}
